com.unity.render-pipelines.universal/Runtime/2D/Light2D.cs

```diff
@@ -293,6 +288,26 @@ private void OnDisable()
             Light2DManager.DeregisterLight(this);
         }
 
+        private int GenerateHashCode()
+        {
+            var hashCode = 0;
+            unchecked
+            {
+                // Spline.
+                hashCode = (int) 2166136261 ^ LightUtility.GetShapePathHash(shapePath);
+                hashCode = hashCode * 16777619 ^ m_FalloffIntensity.GetHashCode();
+                hashCode = hashCode * 16777619 ^ m_ShapeLightFalloffSize.GetHashCode();
+                hashCode = hashCode * 16777619 ^ m_ShapeLightParametricRadius.GetHashCode();
+                hashCode = hashCode * 16777619 ^ m_ShapeLightParametricSides.GetHashCode();
+                hashCode = hashCode * 16777619 ^ m_ShapeLightParametricAngleOffset.GetHashCode();
+                hashCode = hashCode * 16777619 ^ m_Color.GetHashCode();
+                hashCode = hashCode * 16777619 ^ m_LightType.GetHashCode();
+                hashCode = hashCode * 16777619 ^ lightCookieSpriteInstanceID.GetHashCode();
+                hashCode = hashCode * 16777619 ^ m_Intensity.GetHashCode();
+            }
+            return hashCode;
+        }
```

Comment:
Overriding GetHashCode is not possible with MonoBehaviors as Engine uses it to identify as PPtr Instance ID.